### DISCUSSION TITLE
Tracking JWT authorisation usage

### DISCUSSION THREAD

**Discussion by 3goats** [hasura/graphql-engine#8931] [General] [answered]
Hi,

I'm trying to determine if there's an obvious way to track successful JWT authorisations in Hasura. Essentially, I need to update a table upon a successful validation so I can track how many times a JWT has been used so I can limit the number of times it can be used though the permissions builder. Anybody done something like this or any advice ? Ideally, I would like to do it within Hasura without building actions.

**Comment by adas98012** [comment, marked as answer]
One way to authenticate (and gather metadata about requests) is to use auth webhooks. 
Here is the Hasura Doc pertaining to webhook:

https://hasura.io/docs/latest/auth/authentication/webhook/
